### ISSUE TITLE
Repeated authentication request for remote repository

### ISSUE THREAD

**Issue by HiroyasuNishiyama** [node-red/node-red#2545]
### What are the steps to reproduce?
1. prepare repositories
    A. create private repositories on GitHub: Repo-A and Repo-B.
    B. create Node-RED project with remote repository.
    C. commit and push flow to Repo-A.
2. create a Node-RED project from the Repo-A with HTTPS access (user+password).
3. from *Projects*>*Project Settings*>*Settings*>*add remote* add the remote repository Repo-B with HTTPS access.
4. from *history*>*Commit History*>*Manage remote branch* button, select *Remote: origin/master*

### What happens?

Following is repeated:

1. Dialog box for inputting *Username*/*Password} is shown.
2. input user name and password then press *Retry*

![スクリーンショット 2020-04-29 13 40 24](https://user-images.githubusercontent.com/30289092/80561319-045c5280-8a1f-11ea-8d75-99d647e7fb4d.png)

### What do you expect to happen?

A username and password will be accepted at the first input.
However, my understanding of what kind of operation is expected here for multiple remotes is 
ambiguous.

From an implementation point of view,  the reason for this seems to be that the authentication information for the second remote is not properly handled when refreshing information of the repositories.

Similar issue: https://github.com/node-red/node-red/issues/2468

### Please tell us about your environment:

- [x] Node-RED version: 1.0.6
- [x] Node.js version: 12.16.1/12.16.3
- [x] npm version:6.13.4/6.11.4
- [x] Platform/OS: macOS/Ubuntu
- [x] Browser: Chrome

**Comment by knolleary** [comment]
I have managed to track this issue down.

When you click the *manage remote branch* box, we do a `git fetch` of each remote to get the list of remote branches.

If one of those fails due to authentication, we pass an error back to the editor so it can prompt for auth details.

However, we were failing to include information on *which* remote hit the auth error. This meant the editor would prompt for auth details on the first remote in the list - even if that wasn't the remote at fault.

The editor already had the code needed to prompt for the right remote - it just needed to be told which remote to prompt for.

The fix was to ensure the `remote` property of the error was included in the response to the editor.

I have pushed this to `dev` for 1.1.0, but it could be back-ported to `master` in case we do a 1.0.7 release.
